### PR TITLE
Release summary for v2023.06.0

### PR DESCRIPTION
**Pull request by TomNicholas**
Release summary:

This release adds features to ``curvefit``, improves the performance of concatenation, and fixes various bugs.

---

For some reason when I try to use 
`git log "$(git tag --sort=v:refname | tail -1).." --format=%aN | sort -u | perl -pe 's/\n/$1, /'`
to return the list of all contributors since last release, it only returns Deepak :laughing: I'm not sure what's going wrong there - I definitely have all the git tags fetched, and other people have definitely contributed since the last version!

### CODE REVIEW

**Comment by keewis** [comment]
> it only returns Deepak

I'm sorry, I pushed a alpha tag to check whether using the trusted publisher to upload to TestPyPI works and forgot to remove it afterwards. I deleted the tag just now, but you'll have to delete it manually from your repository for the command to work properly.

**Comment by TomNicholas** [comment]
That fixed it! Thanks @keewis

File: doc/whats-new.rst [hunk 1]
```diff
@@ -15,11 +15,45 @@ What's New
     np.random.seed(123456)
 
 
-.. _whats-new.2023.05.1:
+.. _whats-new.2023.06.1:
 
-v2023.05.1 (unreleased)
+v2023.06.0 (unreleased)
```
Comment (dcherian) [inline doc/whats-new.rst:20]: Usually we do this in a separate commit so RTD (which points to the latest tag) doesn't show the WIP changelog

**Comment by TomNicholas** [comment]
Seeing test failures with pint specifically on the python 3.11 build

https://github.com/pydata/xarray/actions/runs/5338204541/jobs/9675258469#step:9:1780

@keewis you think it's fine to ignore that for now?

**Comment by keewis** [comment]
Not sure? The issue seems to be restricted to windows, so not sure why exactly this is failing, and I don't have the time to debug right now.
